torch/_functorch/aot_autograd.py

```diff
@@ -2812,9 +2812,25 @@ def call_compiled_backward():
                 if CompiledFunction.compiled_bw is None:
                     assert all(a is not None for a in all_args)
                     context = disable_autocast_manager if disable_amp else nullcontext
-                    with tracing(saved_context), context(), track_graph_compiling(aot_config, "backward"):
+
+                    placeholder_list = fx_placeholder_vals(bw_module)
+
+                    # We can not reuse tracing context and fake tensors from
+                    # backward pass if any real tensor has a different stride.
+                    # This can happen if the compiler compiles the fwd module
+                    # with layout optimization.
+                    if any(ph_arg.stride() != real_arg.stride()
```

Comment:
Hmm, the reason I add this is, placeholder_list is generated assuming eager mode's behavior (i.e. call make_fx on the joint graph and then do partition). But the compiler may make different decision on activation tensors' layout. This can cause the layouts in `placeholder_list` (eager behavior) mismatch with the layouts in `all_args` (compiler behavior) .

If we pass in `placeholder_list` to the compiler, then the strides we get at compile time may mismatch with the strides we get at runtime. Luckily we have some assertions and fail early (rather than proceed silently and making it hard to debug).